SCClassLibrary/Common/Control/Buffer.sc

```diff
@@ -352,20 +352,18 @@ Buffer {
 	}
 
 	free { arg completionMessage;
+		if(bufnum.isNil) { Error("Cannot call % on a % that has been freed".format(thisMethod.name, this.class.name)).throw };
```

Comment:
I think, in this special case, it's sufficient to ignore the call silently. `free` is not parallel with e.g. `set`.

&nbsp; | `set` | `free`
--|--|--
Desired resulting state | Buffer exists, filled with given data | Buffer does not exist; bufnum is nil
Actual state if called with nil bufnum | Buffer doesn't exist, oops | Buffer does not exist; bufnum is nil

Calling `set` on an invalid buffer is, indeed, a fatal error because the resulting state is completely different from what is desired.

Calling `free` on an invalid buffer results in a final state that matches what you wanted. So, there's no need for an error.

I believe my initial PR handled this correctly.

Reply:
Seems to me that treating free/freeMsg as a special case is what led to this mistake. I'd rather tell the user that they are trying to go something to a nil buffer in a consistent way in all cases. I agree that the error is less severe in the case of free. However, it is still beneficial to the user to be told that they are mistaken about the state of things.

Reply:
I see your point, but disagree.

SC is first and foremost a programming environment for *artists*. Thus (in my view), the overriding concern should not be whether it is strictly correct from a computer-science point of view, but rather whether it supports a creative flow.

`Error().throw` halts execution. That's a disruption to the creative flow. IMO halting execution should be reserved for error conditions that truly are fatal -- where the result is so different from the expectation that it's impossible to recover (or where recovery requires additional logic, which goes into the "catch" branch of a `try` block). Of course, that covers most error scenarios! But otherwise, IMO it's better to get out of the user's way and not distract the user.

I'm observing here a subtle shift in philosophical orientation, one which has not been discussed and for which consensus has not been built, but which is simply "happening" by default. I don't like it.

I'm sorry to be stubborn, but do understand that it's not for the sake of stubbornness. It's because I believe SC should be a creative tool, and it should be fun, and it shouldn't slap the user's wrist over minutiae.

Reply:
@jamshark70 every week, we ask the developers to join a conference call to discuss the state and general direction of the project. It would be amazing if you could join and air these concerns there in more detail so we can discuss it as a group. If you can't or simply don't want to, then we need to find some way to make sure you are being heard. You clearly have very strong and articulate feelings about this, and I don't want development to keep plugging along without your input.

Reply:
I'm a convert. I agree that this should be a warning, and not an error. I'll fix the PR in a bit.